frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -262,6 +234,86 @@ function DataFrame({
   const { columns, sortColumn, getOriginalIndex, getCellContent } =
     useColumnSort(originalNumRows, originalColumns, getOriginalCellContent)
 
+  const applySelections = React.useCallback(
+    // Use debounce to prevent rapid updates to the widget state.
+    debounce(DEBOUNCE_TIME_MS, (newSelection: GridSelection) => {
+      // If we want to support selections also with in the editable mode,
+      // we would need to integrate the `applyEdits` and `applySelections` functions
+      // into a single function that updates the widget state with both the editing
+      // state and the selection state.
+
+      const selectionState = {
+        // We use snake case here since this is the widget state
+        // that is sent and used in the backend. Therefore, it should
+        // conform with the Python naming conventions.
+        select: {
+          rows: [] as number[],
+          columns: [] as string[],
+        },
+      }
+
+      selectionState.select.rows = newSelection.rows.toArray().map(row => {
+        return getOriginalIndex(row)
+      })
+      selectionState.select.columns = newSelection.columns
+        .toArray()
+        .map(columnIdx => {
+          return getColumnName(originalColumns[columnIdx])
+        })
+      const newWidgetState = JSON.stringify(selectionState)
+      const currentWidgetState = widgetMgr.getStringValue(
+        element as WidgetInfo
+      )
+
+      // Only update if there is actually a difference to the previous selection state
+      if (
+        currentWidgetState === undefined ||
+        currentWidgetState !== newWidgetState
+      ) {
+        widgetMgr.setStringValue(
+          element as WidgetInfo,
+          newWidgetState,
+          {
+            fromUi: true,
+          },
+          fragmentId
+        )
+      }
+    }),
+    [widgetMgr, element]
```

Comment:
Does `fragmentId` need to be added to the dependency array?

Reply:
yep, added it to dependencies 👍